project/src/test/java/com/google/sps/CommonUtilsTest.java

```diff
@@ -0,0 +1,62 @@
+import com.google.sps.utilities.CommonUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.springframework.mock.web.MockHttpServletRequest;
+
+/** */
+@RunWith(JUnit4.class)
+public final class CommonUtilsTest {
+  private static final String NAME = "Name";
+  private static final String NAME_VALUE = "Bob";
+  private static final String JOE = "Joe";
+  private static final String JAMES = "James";
+  private static final String[] NONEMPTY_VALUES = {JOE, JAMES};
+  private static final String[] SOME_EMPTY_VALUES = {JOE, ""};
+  private static final String[] EMPTY_VALUES = {"", ""};
+
+  @Test
+  public void getParameterWithValueReturnValue() {
+    MockHttpServletRequest request = new MockHttpServletRequest();
+    request.setParameter(NAME, NAME_VALUE);
+
+    Assert.assertEquals(NAME_VALUE, CommonUtils.getParameter(request, NAME, StringUtils.EMPTY));
+  }
+
+  @Test
+  public void getParameterWithoutValueReturnDefaultValue() {
+    MockHttpServletRequest request = new MockHttpServletRequest();
+
+    Assert.assertEquals(
+        StringUtils.EMPTY, CommonUtils.getParameter(request, NAME, StringUtils.EMPTY));
+  }
+
+  @Test
+  public void getParameterWithNonemptyValuesReturnValues() {
+    MockHttpServletRequest request = new MockHttpServletRequest();
+    request.setParameter(NAME, NONEMPTY_VALUES);
+
+    MatcherAssert.assertThat(
+        CommonUtils.getParameterValues(request, NAME), Matchers.contains(JOE, JAMES));
```

Comment:
When I read the test, I have no idea JOE and JAMES are in the NONEMPTY_VALUES until I read the constants. This is against the best practice.

https://docs.google.com/document/d/1M9kKw-Qgwmi-fhIDab-HCsK6sp53qRJQaM7aM3s4rXs/edit#heading=h.cxwr9nos46ee

I think you can remove the NONEMPTY_VALUES. It is true that we prefer to use constants in production codes, but it is not the same for testing codes.

Reply:
Done - I see, will keep this in mind.